src/import/saga.ts

```diff
@@ -242,21 +242,26 @@ function* attemptBackupPhraseCorrection(mnemonic: string) {
  */
 function* walletHasBalance(address: string) {
   Logger.debug(TAG + '@walletHasBalance', 'Checking account balance')
-  let requests = [
-    yield* fork(fetchTokenBalanceInWeiWithRetry, Currency.Euro, address),
-    yield* fork(fetchTokenBalanceInWeiWithRetry, Currency.Dollar, address),
-    yield* fork(fetchTokenBalanceInWeiWithRetry, Currency.Celo, address),
-  ]
-  while (requests.length > 0) {
-    const balances = yield* race(requests.map((req) => join(req)))
-    for (const balance of balances) {
-      if (balance?.isGreaterThan(0)) {
-        // Cancel any remaining requests.
-        cancel(requests)
-        return true
-      }
-    }
-    requests = requests.filter((_, i) => balances[i] === undefined)
+
+  const getBalanceCEur = yield* fork(fetchTokenBalanceInWeiWithRetry, Currency.Euro, address)
+  const getBalanceCUsd = yield* fork(fetchTokenBalanceInWeiWithRetry, Currency.Dollar, address)
+  const getBalanceCelo = yield* fork(fetchTokenBalanceInWeiWithRetry, Currency.Celo, address)
+
+  const { balanceCEur, balanceCUsd, balanceCelo } = yield* race({
```

Comment:
I guess we don't want to wait the 3 threads to finish if we already know the user has some balance, but I don't know how much we are saving with all this code... Actually, with the current version I think we might be re-running the `fetchTokenBalanceInWeiWithRetry` if they didn't finish at the same time...

I think I prefer to keep the code simpler and wait until the 3 threads finish... What do you think?

Also, this code looks already old, since it only checks for Celo, cUSD and cEUR...

I know this is not part of the RN upgrade, so maybe we should just leave as it was and fix it later? I think we should include other tokens here, this is use in the `importBackupPhraseSaga`, I think we the current code we might be missing some addresses that have other tokens...

Reply:
Yes!! This is even better and the code is so much simpler!!